### ISSUE TITLE
Implement Stickers.

### ISSUE THREAD

**Issue by ProMcTagonist** [qTox/qTox#402]
Emoticons and emoji have been popular almost since instant messaging began. They are small, inline images, that typically replace typed-in, ASCII-character smilies. Emoji are usually differentiated by being represented by the actual unicode characters for what they depict.

qTox currently implements emoticons/emoji by displaying text smilies as graphical ones, and also by offering a panel of graphical smilies that are sent as unicode characters. There is currently one nonstandard graphical smiley in qTox that illustrates an important point: the :Kappa: face.

With emoticons/emoji, you have to worry about what the other person is going to see. This is exacerbated by the free and open nature of Tox. You have no guarantee that other people's clients will display emoticons the same way yours does. This often leads to confusing mishaps, such as emoticons getting inserted where they don't belong, or people not understanding the intended message due to clients using different graphical emoji. The :Kappa: face is exemplary. It's just a joke, and not likely something that other clients will or should implement. But many users WANT to send silly things like that, and have their friends see the intended image. Other users want nothing to do with that, and will be displeased if faces pop up in unrelated text (which does happen, no matter how obscure the replacement strings try to be.)

Emoticons and emoji are from the days of singular, exclusive clients, tied to singular, exclusive servers. Tox breaks free from this paradigm, and this makes it uniquely suited to the new paradigm that many other popular messaging services have already embraced: stickers.

**Stickers** are inline images that are much larger than single-character emoticons or emoji. They can be animated, and for our purposes they could even be webms. The main difference is that stickers are files that get sent. There's no worrying- your friend will see exactly what you mean to send. We don't need closed systems to regulate emoticon replacement, and all the hassle that causes, and we have plenty of bandwidth to send proper images now. It's time that we do so.

Stickers are the perfect platform to demonstrate the benefits of openness and freedom in a very immediate and concrete way. We know that users love stickers on closed systems like Facebook and Path, and with a freedom-oriented system like Tox, everyone can easily make and share their own, fostering the sort of incredible creative community that makes something more than just another app.

To recap:
- Stickers solve problems that arise when emoticons are used for more than they're meant for. Emoticons are great for standard smilies; stickers are right for everything else.
- Stickers cure headaches caused by string replacement and unicode problems.
- Stickers are a feature that users love, and will prove to be just another bonus of Tox.
- Stickers can show average users why freedom is better.

I suggest copying Facebook's sticker interface implementation, as it's clean, well-known, and working, and then making improvements where needed through further FOSS evolution.
Thanks for reading.

**Comment by tux3** [comment]
That's an interesting idea, but I <a href="https://github.com/irungentoo/toxcore/search?utf8=%E2%9C%93&q=sticker">don't think stickers are implemented in toxcore yet</a>.

You should probably open an issue on toxcore first, it's technically possible to implement that on qTox only, but then other clients wouldn't see them, which isn't really better than what we have with emoticons/emoji.

**Comment by ProMcTagonist** [comment]
irungentoo has commented in the past that "stickers are bloat," but I'll make an issue over on core.
There's been some contention lately over what should and shouldn't be in the core, and I'll leave it up to developers to decide where this should be (I had been under the impression that there was a way clients could do this themselves with custom packets, apologies).

**Comment by ProMcTagonist** [comment]
https://github.com/irungentoo/toxcore/issues/1106

**Comment by dubslow** [comment]
This is a nice feature, but is something to implement near the end of beta, when the core features work.

(Also did anyone even notice that I added :Kappa:? Sorry but I love that image ...:Kappa:)

**Comment by tux3** [comment]
> I had been under the impression that there was a way clients could do this themselves with custom packets, apologies

This is indeed possible, but unless all clients agree to add this, it's not going to work. And I'd rather avoid using custom packets as much as possible, for compatibility reasons.

**Comment by ProMcTagonist** [comment]
> (Also did anyone even notice that I added :Kappa:? Sorry but I love that image ...:Kappa:)

I noticed some groupchat people appreciating him :) I was confused before I updated qTox, because when I think kappa I think http://i.imgur.com/UEeWluQ.png

> And I'd rather avoid using custom as much as possible, for compatibility reasons.

That's understandable. Hopefully the core issue gets some discussion going among the people in charge of it there.

**Comment by ProMcTagonist** [comment]
Stickers can be implemented with FILE_KIND, yes?

**Comment by zetok** [comment]
> Stickers can be implemented with FILE_KIND, yes?

~kinda.
